test/verify/check-system-realms

```diff
@@ -279,6 +279,8 @@ class CommonTests:
         # (this should be fixed properly in cockpit-ws)
         self.allow_journal_messages('.*session timed out during authentication')
         self.allow_journal_messages('.*authentication timed out')
+        # 14 == SIGALRM
+        self.allow_journal_messages('.*bridge program failed: Child process killed by signal 14')
```

Comment:
Argh, yes, I didn't pay enough attention to this :see_no_evil:  This is doubly nonsense, as the *bridge* gets killed with that alarm, not cockpit -session. I. e. that alarm(60) somehow gets inherited.

I see one obvious bug here, there is a `signal (SIGALRM, SIG_DFL);` call later on which I missed. However, after fixing that, I still get that error. It almost looks like starting a PAM session inherits the current alarms and signal handlers, and that of course is really not intended.

This confirms my initial skepticism about making the range of that alert(60) so wide. I reverted it to a narrower scope of waiting for and parsing the initial authorize dialog.

Reply:
I think our original issue here is that this message is misleading.  This journal message is coming from `-ws` (`cockpitpipetransport.c:125`), which prints it after observing the exit status of `cockpit-session`.

The child doesn't inherit `alarm()` across a `fork()`.  This is the documented behaviour per [POSIX](https://pubs.opengroup.org/onlinepubs/009695399/functions/alarm.html), [Linux](https://man7.org/linux/man-pages/man2/fork.2.html), and I tested that it's true.

Somehow the `-session` process itself is the thing that's getting killed by the signal.  Which means that there's still something unexplained going on here.

Reply:
Yes, as said on Matrix it's *definitively* not our `spawn_and_wait()`. That came after the alarm disarming even in the previous version of the patch. Hence my suspicion that it's the PAM session start, because what else could it be?

Reply:
Note that I did test the signal handler, with alarm(1) and an extra sleep in ws (I put the patch somewhere, but can't find it quickly -- too many threads). That worked as expected, including the correct message.

Reply:
Also note: This piece of the patch is entirely gone now.

Reply:
I've tried to chase this down long enough to have thoroughly disproven my own original theory of what was going on here and to have (re-)learned with a high degree of certainty exactly what the process model is here.

The tl;dr: of it all is that a million things use `alarm()` for all kinds of reasons, and many of them attempt to restore the previous state.  I've been chasing ghosts through glibc with the locking functions in the utmp code and the password database and haven't been able to come up with anything, but I'm convinced enough at this point that the culprit is definitely in-process.

I also don't care anymore.  It would have been nice to expand this to cover more code, but my concerns about the user being able to delay the other parts of the code are completely theoretical, and definitely "not our bug".  Thanks for putting up with me on this.  I'm not satisfied with the (lack of) answer but I'm sure enough about it to know that nothing deeply "weird" is happening here, and even if it is, it's not our fault.